examples/sagaMonitor/modules/Manager.js

```diff
@@ -0,0 +1,30 @@
+/** The manager is used for bookkeeping all the effect descriptors */
+export default class Manager {
+  constructor() {
+    this.rootIds = []
+    // effect-id-to-effect-descriptor
+    this.map = {}
+  }
+
+  get(effectId) {
+    return this.map[effectId]
+  }
+
+  set(effectId, desc) {
+    this.map[effectId] = desc
+  }
+
+  setAsRoot(effectId) {
+    this.rootIds.push(effectId)
+  }
+
+  getRootIds() {
+    return this.rootIds
+  }
+
+  getChildIds(parentEffectId) {
+    return Object.keys(this.map)
+      .filter(effectId => this.map[effectId].parentEffectId === parentEffectId)
+      .map(Number)
```

Comment:
is this necessary? effect ids should be numbers already, right?

Reply:
Keys of object are always strings, and `Object.keys(obj)` returns an array of strings.
![image](https://user-images.githubusercontent.com/5550931/45045134-adb14080-b0a4-11e8-8205-82d5687b54e0.png)

Reply:
oh, yeah - havent quite noticed they are coming here from Object.keys